server/text_generation_server/models/flash_llama.py

```diff
@@ -35,9 +35,13 @@ def __init__(
         if torch.cuda.is_available():
             device = torch.device(f"cuda:{rank}")
             dtype = torch.float16 if dtype is None else dtype
-        elif IS_XPU_SYSTEM:
-            device = torch.device(f"xpu:{rank}")
-            dtype = torch.float16 if dtype is None else dtype
+        elif IS_IPEX_AVAILABLE:
+            if IS_XPU_SYSTEM:
+                device = torch.device(f"xpu:{rank}")
+                dtype = torch.float16 if dtype is None else dtype
+            else:
+                device = torch.device("cpu")
```

Comment:
IS flash attention + paged implemented for IPEX + CPU ? 
Or any CPU ?

Reply:
for IPEX+CPU, IPEX provide same pageattention+flash attention op for xpu+cpu